core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala

```diff
@@ -75,9 +75,11 @@ class KryoSerializerSuite extends SparkFunSuite with SharedSparkContext {
   }
 
   test("basic types") {
-    val ser = new KryoSerializer(conf).newInstance()
     def check[T: ClassTag](t: T) {
-      assert(ser.deserialize[T](ser.serialize(t)) === t)
+      testBothUnsafeAndSafe(conf) { sparkConf =>
```

Comment:
Is there any way to refactor this into two Suites, where one extends from the other and has additional setup code (or a setup method that it overrides)? Otherwise we have to remember to add this construct to all tests.